projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-list/query-list.component.ts

```diff
@@ -1,18 +1,90 @@
-import { Component, Input, OnInit } from '@angular/core';
-import { PartyMessages } from '../../domain';
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { SortOrder } from '../../../complex/sort-order';
+import { PartyMessagesGroup, QueryListData, column } from '../../domain';
 
 @Component({
   selector: 'ccd-query-list',
   templateUrl: './query-list.component.html',
+  styleUrls: ['./query-list.component.scss']
 })
-export class QueryListComponent implements OnInit {
+export class QueryListComponent implements OnChanges {
+  @Input() public partyMessageGroup: PartyMessagesGroup;
+  public queryListData: QueryListData | undefined;
+  public displayedColumns: column[] = [
+    { name: 'subject', displayName: 'Queries', sortOrder: SortOrder.UNSORTED },
+    { name: 'lastSubmittedBy', displayName: 'Last submitted by', sortOrder: SortOrder.UNSORTED },
+    { name: 'lastSubmittedDate', displayName: 'Last submission date', sortOrder: SortOrder.UNSORTED },
+    { name: 'lastResponseDate', displayName: 'Last response date', sortOrder: SortOrder.UNSORTED },
+    { name: 'lastResponseBy', displayName: 'Response by', sortOrder: SortOrder.UNSORTED }
+  ];
 
-  @Input() public partyMessages: PartyMessages[];
+  public ngOnChanges(simpleChanges: SimpleChanges) {
+    const currentPartyMessageGroup = simpleChanges.partyMessageGroup?.currentValue as PartyMessagesGroup;
+    if (currentPartyMessageGroup) {
+      this.queryListData = new QueryListData(currentPartyMessageGroup);
+    }
+  }
+
+  public sortTable(col: column) {
+    switch (col.displayName) {
+      case 'Queries': {
+        this.sort(col);
+        break;
+      }
+      case 'Last submitted by': {
+        this.sort(col);
+        break;
+      }
+      case 'Last submission date': {
+        this.sortDate(col);
+        break;
+      }
+      case 'Last response date': {
+        this.sortDate(col);
+        break;
+      }
+      case 'Response by': {
+        this.sort(col);
+        break;
+      }
+    }
+  }
 
-  constructor() {
+  public sortWidget(col: column): string {
+    switch (col.sortOrder) {
+      case SortOrder.DESCENDING: {
```

Comment:
I think the order should be different here;

DESCENDING should be a down triangle
ASCENDING should be an up triangle (smaller at the top and increasing as it goes to bottom).

Reply:
changed